src/ompl/geometric/planners/rrt/STRRTstar.h

```diff
@@ -0,0 +1,547 @@
+            {
+                optimize_ = optimize;
+            }
+
+            /** \brief Get whether the planner optimizes the solution. */
+            bool getOptimize() const
+            {
+                return optimize_;
+            }
+
+            double getOptimumApproxFactor() const
+            {
+                return optimumApproxFactor_;
+            }
+
+            void setOptimumApproxFactor(double optimumApproxFactor)
+            {
+                if (optimumApproxFactor <= 0 || optimumApproxFactor > 1)
+                {
+                    OMPL_ERROR("%s: The optimum approximation factor needs to be between 0 and 1.", getName().c_str());
+                }
+                optimumApproxFactor_ = optimumApproxFactor;
+            }
+
+            std::string getRewiringState() const
+            {
+                std::vector<std::string> s{"Radius", "KNearest", "Off"};
+                return s[rewireState_];
+            }
+
+            void setRewiringToOff()
+            {
+                rewireState_ = OFF;
+            }
+
+            void setRewiringToRadius()
+            {
+                rewireState_ = RADIUS;
+            }
+
+            void setRewiringToKNearest()
+            {
+                rewireState_ = KNEAREST;
+            }
+
+            double getRewireFactor() const
+            {
+                return rewireFactor_;
+            }
+
+            void setRewireFactor(double v)
+            {
+                if (v <= 1)
+                {
+                    OMPL_ERROR("%s: Rewire Factor needs to be greater than 1.", getName().c_str());
+                }
+                rewireFactor_ = v;
+            }
+
+            unsigned int getBatchSize() const
+            {
+                return initialBatchSize_;
+            }
+
+            void setBatchSize(int v)
+            {
+                if (v < 1)
+                {
+                    OMPL_ERROR("%s: Batch Size needs to be at least 1.", getName().c_str());
+                }
+                initialBatchSize_ = v;
+            }
+
+            void setTimeBoundFactorIncrease(double f)
+            {
+                if (f <= 1.0)
+                {
+                    OMPL_ERROR("%s: Time Bound Factor Increase needs to be higher than 1.", getName().c_str());
+                    OMPL_ERROR("%s: Time Bound Factor Increase needs to be higher than 1.", getName().c_str());
+                }
+                timeBoundFactorIncrease_ = f;
+            }
+
+            void setInitialTimeBoundFactor(double f)
+            {
+                if (f <= 1.0)
+                {
+                    OMPL_ERROR("%s: Initial Time Bound Factor Increase needs to be higher than 1.", getName().c_str());
+                }
+                initialTimeBoundFactor_ = f;
+            }
+
+            void setSampleUniformForUnboundedTime(bool uniform)
+            {
+                sampleUniformForUnboundedTime_ = uniform;
+            }
```

Comment:
All the Getter and Setter method clutter a bit the header file (and it would take a longer time to recompile if you change something here). I would move the implementation to the cpp file to have a cleaner hpp file.